src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -646,41 +648,163 @@ sub check_dither {
+    my $dthr = shift;
+    my %standard_y_dither = (20 => 1087.0,
+                             8 => 1000.0);
+    my %standard_z_dither = (20 => 768.6,
+                             8 => 707.1);
+
+    # If the rounded amplitude is not in the standard set, return 0
+    if (not (grep $_ eq int($dthr->{ampl_p} + .5), (keys %standard_z_dither))){
+        return 0;
+    }
+    if (not (grep $_ eq int($dthr->{ampl_y} + .5), (keys %standard_y_dither))){
+        return 0;
+    }
+    # If the period is not standard for the standard amplitudes return 0
+    if (abs($dthr->{period_p} - $standard_z_dither{int($dthr->{ampl_p} + .5)}) > 10){
+        return 0;
+    }
+    if (abs($dthr->{period_y} - $standard_y_dither{int($dthr->{ampl_y} + .5)}) > 10){
+        return 0;
+    }
+
+    # If those tests passed, the dither is standard
+    return 1;
+}
+
+
+#############################################################################################
+sub large_dither_checks {
+#############################################################################################
+    # Check the subset of monitor-window-style commanding that should be used on
+    # observations with large dither.
+
+    my $self = shift;
+    my $dither_state = shift;
+    my $all_dither = shift;
+    my $time_tol = 10;         # Commands must be within $time_tol of expectation
+
+    # Now check in backstop commands for :
+    #  Dither is disabled (AODSDITH) 1 min prior to the end of the maneuver (EOM)
+    #    to the target attitude.
+    #  Dither is enabled (AOENDITH) 5 min after EOM
+    #  Dither parameters are commanded 5 min before end of observation
+    # ACA-040
+    # obs_tstart is defined as the tstop of the maneuver to this observation in set_npm_times
+    # obs_tstop is defined as the time of the maneuver away or the end of the schedule
+
+    # Is the large dither command enabled 5 minutes after EOM?
+    if (abs($dither_state->{time} - $self->{obs_tstart} - 300) > $time_tol){
+        push @{$self->{warn}},
+            sprintf("$alarm Large Dither not enabled 5 min after EOM (%s)\n",
+                    time2date($self->{obs_tstart}));
+    }
+    # What's the dither state at EOM?
+    my $obs_start_dither;
+    foreach my $dither (reverse @{$all_dither}) {
+	if ($self->{obs_tstart} >= $dither->{time}) {
+            $obs_start_dither = $dither;
+            last;
+        }
+    }
+
+    # Is dither disabled at EOM and one minute before?
+    if ((abs($self->{obs_tstart} - $obs_start_dither->{time} - 60) > $time_tol)
+            or ($obs_start_dither->{state} ne 'DISA')){
+        push @{$self->{warn}},
+            sprintf("$alarm Dither should be disabled 1 min before obs start for Large Dither\n");
+    }
+
+    # Are parameters set to normal values 5 minutes before the end of the observation?
+    my $obs_stop_dither;
+    foreach my $dither (reverse @{$all_dither}) {
+	if ($self->{obs_tstop} - 300 >= $dither->{time}) {
```

Comment:
This appears to be checking for any dither commanding that happens earlier than 5 mins before `obs_tstop`, not for commanding that occurs within some narrow window.  Maybe in conjunction with the logic of other tests that is sufficient, but it makes the tests more difficult to verify analytically.